dockerfiles/theia/Dockerfile

```diff
@@ -33,50 +33,31 @@ ENV THEIA_VERSION=${THEIA_VERSION} \
     HOME=/home/theia
 WORKDIR ${HOME}
 
-# Add verdaccio config
-ADD src/config.yaml ${HOME}/.config/verdaccio/
+# add eclipse che theia generator
+RUN yarn global add @eclipse-che/theia-generator@0.0.1-1543478099
 
 # Add patches
 ADD src/patches /home/theia-build/patches
 
 # Invalidate cache if there is a new commit
 ADD https://api.github.com/repos/theia-ide/theia/git/refs/tags/v${THEIA_VERSION} /tmp/sha1-repository
 # Build theia from source code and publish artifacts locally
-ADD src/builder/build-and-publish-theia.sh /tmp/builder/scripts/build-and-publish-theia.sh
-RUN /tmp/builder/scripts/build-and-publish-theia.sh
+ADD src/builder/clone.sh /tmp/builder/scripts/clone.sh
+RUN /tmp/builder/scripts/clone.sh
```

Comment:
Would it be ok to (In this PR)
-  create a "dockerfiles/theia-builder" with the builder + theses scripts (somewhere else then /tmp/, in the $PATH maybe)
- In here (multistage build) we would just be based on eclipse/che-theia-builder and call the scripts (clone.sh, yarn production.sh etc ...
WDYT ?

Reply:
it will be another PR (to have another image for the build tools)
but IMHO in any case it should be a multi-staged build to be sure to have the less layer than possible for the runtime (the smallest image)
it's not a multi-staged build, it means that all actions will be done again each time you change anything

Reply:
BTW I also think that we don't need sh scripts in builder image. (can be discussed there)
but the 'npm script' is handling it (or should handle other stuff)

Reply:
yes but at some point you may also want to try a 'production' build in che and run it in a production container without having to rebuild it (just share the right volume)

Reply:
> it will be another PR (to have another image for the build tools)
OK

> but IMHO in any case it should be a multi-staged build to be sure to have the less layer than possible for the runtime (the smallest image)
> it's not a multi-staged build, it means that all actions will be done again each time you change anything
Yes agree. We would keep the multistaged build:
```
FROM eclipse/che-theia-builder as builder
..... run scripts clone, etc ...
....